File: 1ºAno/Lista3/Python/Sala/files/ex2.py
while True:
    while True:
        try:
            a1, an, r = input("Na mesma linha, entre com os valores, respectivamente, referentes a:\n\n1) Primeiro valor da sequência (a1)\n2) Último valor da sequência (an)\n3) Razão da sequência (r)\nPara sair, entre com 0 em todos os valores\n\n").split()
            a1, an , r = int(a1), int(an), int(r)
            break
        except:
            print ("Número(s) inválidos... Tente novamente")
    if a1 == an ==  r == 0:
        break
    if a1 <= an:
        if a1 % r != 0:
            a1 -= a1 % r - r
        if an % r:
            an -= an % r
        print ("Sn = {}\n" .format((a1 + an) * ((an - a1) // 2 + 1)))
    else:
        if a1 % r != 0:
            a1 -= a1 % r
        if an % r:
            an -= an % r - r
        print ("Sn = {}\n" .format((a1 + an) * ((a1 - an) // 2 + 1)))
print ("Adeus")

File: 1ºAno/Lista3/C/Sala/files/ex2.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int main() {
    for (;;) {
        char * dados = (char*)malloc(10 * sizeof (char));
        int vals[3];
        for (;;) {
            printf ("Entre, na mesma linha, com os seguintes valores, respectivamente:\n\n1) 1º número da sequência (a1)\n2) Último número da sequência (an)\n3) Divisibilidade dos termos\nPara sair, entre com 0 em todos os valores\n\n");
            fgets(dados, 10, stdin);
            char *pt = strtok(dados, " ");
            int i = 0;
            while (pt != NULL) {
                vals[i] = atoi(pt);
                pt = strtok(NULL, " ");
                ++i;
            }
            if (vals[0] == 0 && vals[1] == 0 & vals[2] == 0) {
                break;
            }
            if (vals[2] == 0) {
                printf ("Número(s) inválidos... Tente novamente\n");
            }
            else {
                break;
            }
        }
        if (vals[0] == 0 && vals[1] == 0 & vals[2] == 0) {
            break;
        }
        if (vals[0] <= vals[1]) {
            if (vals[0] % vals[2] != 0) {
                vals[0] -= vals[0] % vals[2] - vals[2];
            }
            if (vals[1] % vals[2] != 0) {
                vals[1] -= vals[1] % vals[2];
            }
            printf ("Sn = %d\n\n", (vals[0] + vals[1]) * ((vals[1] - vals[0]) / vals[2] + 1) / 2);
        }
        else {
            if (vals[0] % vals[2] != 0) {
                vals[0] -= vals[0] % vals[2];
            }
            if (vals[1] % vals[2] != 0) {
                vals[1] -= vals[1] % vals[2] - vals[2];
            }
            printf ("Sn = %d\n\n", (vals[0] + vals[1]) * ((vals[0] - vals[1]) / vals[2] + 1) / 2);
        }
    }
    printf ("\nAdeus!");
    return 0;
}